src/tribler/gui/tribler_window.py

```diff
@@ -1254,3 +1254,14 @@ def handle_uri(self, uri):
         self.pending_uri_requests.append(uri)
         if self.tribler_started and not self.start_download_dialog_active:
             self.process_uri_request()
+
+    def show_database_is_corrupted_dialog(self, db_file_path: Path):
+        self.dialog = ConfirmationDialog(
+            self,
+            tr("Database is corrupted: restart needed"),
+            tr("The database file found to be corrupted") + ":<br/>"
+            + f"<b>{db_file_path}</b><br/><br/>"
+            + tr("Please restart Tribler to re-create the database file"),
```

Comment:
Tribler has to be restarted automatically or, at least closed automatically.

Reply:
I completely reimplemented this logic, and now Tribler restarts automatically without requiring action from the user.